ROS2/zenoh-pico-cyclonedds-cdr-message-log/z_pub_ros2.c

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright (c) 2022 ZettaScale Technology
+// Copyright (c) 2022 NXP
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//   Peter van der Perk, <peter.vanderperk@nxp.com>
+//
+
+#include <ctype.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <zenoh-pico.h>
+#include <rcl_interfaces/msg/Log.h>
+
+// CycloneDDS CDR Deserializer
+#include <dds/cdr/dds_cdrstream.h>
+
+// CDR Xtypes header {0x00, 0x01} indicates it's Little Endian (CDR_LE representation)
+const uint8_t ros2_header[4] = {0x00, 0x01, 0x00, 0x00};
+
+const size_t alloc_size = 4096; // Abitrary size 
+
+int main(int argc, char **argv) {
+    const char *keyexpr = "rt/zenoh_log_test";    
+    const char *value = "Pub from Pico IDL!";

```

Comment:
This variable is not used.
I did a PR to remove unused variable and apply formatting.